cEP-0025.md

```diff
@@ -0,0 +1,306 @@
+    """
+    BEAR_DEPS = {PyFlakesASTBear}
+
+    def remove_future_imports(self, file, lineno,
+                              max_col_offset, affected_lines):
+        """
+        Removes all __future__ imports from the input line
+        """
+        diff = Diff(file)
+        line = file[lineno - 1]
+        # If a line contains \ in the end
+        if line.rstrip()[-1] == '\\':
+            next_line = file[lineno]
+            semicolon_index = next_line.find(';')
+            diff.delete_line(lineno)
+            if semicolon_index == -1:
+                diff.delete_line(lineno + 1)
+            else:
+                replacement = next_line[semicolon_index + 1:].lstrip()
+                diff.modify_line(lineno+1, replacement)
+        else:
+            semicolon_indices = [i for i, a in enumerate(line) if a == ';']
+            if len(semicolon_indices) == 0:
+                diff.delete_line(lineno)
+            else:
+                for indice in semicolon_indices:
+                    if indice > max_col_offset:
+                        replacement = line[indice + 1:].lstrip()
+                        diff.modify_line(lineno, replacement)
+                        break
+
+        return diff
+
+    def run(self, filename, file,
+            dependency_results=dict()
+            ):
+        """
+        Uses PyFlakesASTBear to get only FutureImportation
+        """
+        # Preprocess all the lines containing __future__ imports
+        affected_lines = set()
+        # Get the maximum column offset of FutureImportation for a line
+        max_column_for_a_line = dict()
+        for result in dependency_results.get(PyFlakesASTBear.name, []):
+            for node in result.get_nodes(result.module_scope[0],
+                                         FutureImportation):
+                lineno = result.get_node_location(node)[0]
+                col_offset = result.get_node_location(node)[2]
+                if lineno in max_column_for_a_line:
+                    max_column_for_a_line[lineno] = max(
+                                max_column_for_a_line[lineno],
+                                col_offset)
+                else:
+                    max_column_for_a_line[lineno] = col_offset
+                affected_lines.add(lineno)
+
+        for line in affected_lines:
+            corrected = self.remove_future_imports(
+                        file, line, max_column_for_a_line[lineno],
+                        affected_lines
+                        )
+            yield Result.from_values(
+                origin=self,
+                message=('Future import/imports found'),
+                file=filename,
+                diffs={filename: corrected},
+                line=line)
+```
+
+## A generic implementation for detecting `__future__` imports
+
+This implementation doesn't requires coala to be installed and is
+designed so that it can be used in other tools like flake8. This
```

Comment:
clarify that coala will also be able to use these plugins, but these are also usable by flake8.

Also this example should be above the 'fixes' bear.

The fixes bear is the finale -- it shows why we would use coala bears -- there are features of coala which bears can use that generic plugins can not use.